src/query/frontend/semantic/symbol_generator.cpp

```diff
@@ -447,6 +447,14 @@ bool SymbolGenerator::PreVisit(Extract &extract) {
 bool SymbolGenerator::PreVisit(Exists &exists) {
   auto &scope = scopes_.back();
 
+  if (scope.in_set_property) {
+    throw utils::NotYetImplemented("Set property can not be used with exists, but only during matching!");
+  }
+
+  if (scope.in_with) {
+    throw utils::NotYetImplemented("WITH can not be used with exists, but only during matching!");
+  }
```

Comment:
Is this the best option for a symbol generator, shouldn't we throw in a different part of the code? Rule base planer seems too far, maybe preprocess.hpp? I am not sure

Reply:
but it is currently a semantic error, because semantically currently it does not make sense to put exists with WITH clause?

Reply:
yes, true